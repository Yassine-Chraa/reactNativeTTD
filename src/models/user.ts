import UserInterface from "../interfaces/UserInterface"

export const fetchUserDetails = ():Promise<UserInterface> => {
    return Promise.resolve({
        fullName: "Yassine Chraa",
        age: 22
    })
}