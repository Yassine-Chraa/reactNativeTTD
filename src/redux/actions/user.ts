import {Dispatch} from 'react';

export const getUserAction = () => async (dispatch: Dispatch<any>) => {
  try {
    //const {data} = await axios.get(`${usersUrl}/profile`);
    dispatch({type: `Get User`, payload: {FullName: 'Yassine Chraa', age: 22}});
    return {FullName: 'Yassine Chraa', age: 22};
  } catch (error: any) {
    return false;
  }
};
