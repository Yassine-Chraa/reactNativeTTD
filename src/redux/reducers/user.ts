import UserInterface from '../../interfaces/UserInterface';

export default (
  user: UserInterface | false,
  action: {type: String; payload: UserInterface},
) => {
  switch (action.type) {
    case 'Get User':
      return action.payload;
    case 'Update User':
      if (user) {
        return {
          fullName: action.payload.fullName || user.fullName,
          age: action.payload.age || user.age,
        };
      } else {
        return action.payload;
      }
    default:
      return false;
  }
};
