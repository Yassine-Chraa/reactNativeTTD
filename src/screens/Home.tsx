import { Text, View} from "react-native"
import useHomeModel from "../viewmodels/HomeModel";

const Home = ()=>{
    const {userDetails} = useHomeModel();
   
    return(
        <View>
            <Text>FullName: {userDetails?.fullName}</Text>
            <Text>age: {userDetails?.age}</Text>
        </View>
    )
}

export default Home;
