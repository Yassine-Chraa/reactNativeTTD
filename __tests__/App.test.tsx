import React from 'react';
import { it } from '@jest/globals';
import renderer from 'react-test-renderer';
import App from '../App';
import Home from '../src/screens/Home';
import { fetchUserDetails } from '../src/models/user';
import { getUserAction } from '../src/redux/actions/user';
import user from '../src/redux/reducers/user';

describe("Main App", () => {
  it('renders correctly', () => {
    renderer.create(<App />);
  });
})

describe("Home Screen", () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Fetch User Details', () => {
    fetchUserDetails().then((res) => {
      expect(res).not.toBeNull();
    })
  })
  it.todo("...");
})

describe("Redux Reducers", () => {
  it("fetch user details", () => {
    const res = user(false,{type:"Get User",payload:{ fullName: "Yassine Chraa", age: 22 }});
    expect(res).toEqual({ fullName: "Yassine Chraa", age: 22 })
  })
  it("update user",()=>{
    const currentState = { fullName: "Yassine Chraa", age: 22 };
    const res = user(currentState,{type: "Update User",payload:{fullName: "Ahmed Chraa"}})
    expect(res).toEqual({fullName: "Ahmed Chraa",age: 22})
  })
})

