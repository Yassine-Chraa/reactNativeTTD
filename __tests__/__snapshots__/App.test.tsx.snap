// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen renders correctly 1`] = `
<View>
  <Text>
    FullName: 
  </Text>
  <Text>
    age: 
  </Text>
</View>
`;

exports[`Profile Screen renders correctly 1`] = `
<View>
  <Text>
    FullName: 
  </Text>
  <Text>
    age: 
  </Text>
</View>
`;
